presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/JoinMatcher.java

```diff
@@ -97,9 +98,11 @@ public MatchResult detailMatches(PlanNode node, PlanNodeCost cost, Session sessi
     @Override
     public String toString()
     {
-        return toStringHelper(this)
-                .add("equiCriteria", equiCriteria)
-                .add("filter", filter)
-                .toString();
+        ToStringHelper stringHelper = toStringHelper(this)
+                .add("equiCriteria", this.equiCriteria);
+        if (filter.isPresent()) {
```

Comment:
you can do this fluently using `ToStringHelper.omitNulls()`. I think readability is then much better.